### PR TITLE
MAINT: Updates to formatters in `matplotlib.ticker`

### PR DESCRIPTION
**Pull request by madphysicist**
This PR is a split from #6251. Most of the changes are to docs, but there are two minor code changes:

  1. `__all__` has been moved to the top of the file and a couple of missing formatters were added to it.
  2. A second format (position) is permitted in the string for the new-style string formatter.

As the title suggests, only formatters were affected. Locators were left untouched.

I have retained the original commits until the PR is reviewed to make sure that I did the split correctly.

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -240,22 +256,32 @@ def set_bounds(self, vmin, vmax):
...
+        position unspecified.
+        """
         return self.__call__(value)
 
     def format_data_short(self, value):
-        """return a short string version"""
+        """
+        Return a short string version of the tick value.
+
+        Defaults to the position-indepedent long value.
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: independent

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -335,42 +375,62 @@ def set_offset_string(self, ofs):
...
         return self.func(x, pos)
 
 
 class FormatStrFormatter(Formatter):
     """
-    Use an old-style ('%' operator) format string to format the tick
+    Use an old-style ('%' operator) format string to format the tick.
+
+    The format string should have a single variable format (%) in it.
+    It will be applied to the value (not the postition) of the tick.
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: position

File: lib/matplotlib/ticker.py [hunk 3]
```diff
@@ -418,16 +486,20 @@ def pprint_val(self, x, d):
...
-    such that the tick labels are meaningful. Scientific notation is used for
-    data < 10^-n or data >= 10^m, where n and m are the power limits set using
-    set_powerlimits((n,m)). The defaults for these are controlled by the
-    axes.formatter.limits rc parameter.
-
+    Format tick values as a number.
     """
-
     def __init__(self, useOffset=None, useMathText=None, useLocale=None):
+        """
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: Does it make sense to document both class and `__init__`? I think Sphinx has a setting for it, so it may ignore one or the other, or even just combine both together.

File: lib/matplotlib/ticker.py [hunk 4]
```diff
@@ -487,33 +561,45 @@ def __call__(self, x, pos=None):
...
-        '''
+        """
         Sets size thresholds for scientific notation.
 
-        e.g., ``formatter.set_powerlimits((-3, 4))`` sets the pre-2007 default
-        in which scientific notation is used for numbers less than 1e-3 or
-        greater than 1e4.
-        See also :meth:`set_scientific`.
-        '''
+        Limits is a two-element sequence containing the powers of 10
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: What's Limits? There's no argument with that name.

File: lib/matplotlib/ticker.py (hunk 3 above)
Comment (madphysicist) [inline lib/matplotlib/ticker.py]: Good point. Merged into class description. It is more likely to get picked up and put in the right place in case there is a difference.

File: lib/matplotlib/ticker.py [hunk 5]
```diff
@@ -688,29 +777,38 @@ def _formatSciNotation(self, s):
...
+        """
         self._base = base
 
     def label_minor(self, labelOnlyBase):
-        'switch on/off minor ticks labeling'
+        """
+        Switch minor tick labeling on or off.
+
+        ``labelOnlyBase=True`` to turn off minor ticks.
+        """
```
Comment (tacaswell) [inline lib/matplotlib/ticker.py]: Does this currently work?  Should it have a `not` in there?

Comment (madphysicist) [inline lib/matplotlib/ticker.py]: This particular method I have no idea. The formatter itself seems to work OK though.

Comment (madphysicist) [inline lib/matplotlib/ticker.py]: This method does nothing for this particular class since `self.labelOnlyBase` is not used in `self.pprint_val`. However, it does work for the `LogFormatterExponent` class, which extends this one. The old docs and the method name are indeed very misleading. `labelOnlyBase` is named correctly (as I have noted in the updated docs). If it is `True`, minor ticks will *not* be shown, which is exactly the opposite of what the name of the method implies. I did not find any explicit calls to this method in `matplotlib`. Am I free to modify the parameter name to say `labelMinor` and set `self.labelOnlyBase = not labelMinor`?

Comment (tacaswell) [inline lib/matplotlib/ticker.py]: Yes, but can you do that in a separate PR?  The doc updates are non-controversial and should go in, but the API change should get more attention.

Comment (madphysicist) [inline lib/matplotlib/ticker.py]: Makes sense, will do.

Comment (madphysicist) [inline lib/matplotlib/ticker.py]: I have opened up https://github.com/matplotlib/matplotlib/pull/6264 with the code change.

**Comment by tacaswell** [comment]
This looks good, but needs a rebase :disappointed: \

**Comment by madphysicist** [comment]
Fixed. I've been spoiled by Homu on numpy.

**Comment by tacaswell** [comment]
backported to v2.x as 668f292b3c131f2eafbd89e351200937ac5bc87c
